core/experiments/assets/js/admin/behaviors/experiments-messages.js

```diff
@@ -25,19 +25,26 @@ export default class ExperimentsDependency {
 		} );
 	}
 
+	inactiveStateActivity( experimentId ) {
+		if ( !! this.isExperimentContainsDeactivatingMessage( experimentId ) && elementorAdminConfig.experiments[ experimentId ].state !== STATE_INACTIVE ) {
+			this.showDialog( experimentId, STATE_INACTIVE );
+		} else {
+			this.deactivateDependantExperiments( experimentId );
+		}
+	}
 	onExperimentStateChange( e ) {
 		const { experimentId } = e.currentTarget.dataset,
 			experimentNewState = this.getExperimentActualState( experimentId );
 
 		switch ( experimentNewState ) {
 			case STATE_ACTIVE:
 				if ( this.shouldShowDependenciesDialog( experimentId ) ) {
-					this.showDependenciesDialog( experimentId );
+					this.showDialog( experimentId, STATE_ACTIVE );
 				}
 				break;
 
 			case STATE_INACTIVE:
-				this.deactivateDependantExperiments( experimentId );
+				this.inactiveStateActivity( experimentId );
```

Comment:
use the same convention as the previous case (if applicable)

Reply:
refactored to `shouldShowDeactivateModal`